WordPress/src/main/java/org/wordpress/android/models/Note.java

```diff
@@ -482,6 +482,21 @@ public JSONArray getHeader() {
         }
     }
 
+    // this method is used to compare two Notes: as it's potentially a very processing intensive operation,
+    // we're only comparing the note id, timestamp, and raw JSON length, which is accurate enough for
+    // the purpose of checking if the local Note is any different from a remote note.
+    public boolean similar(Note note) {
```

Comment:
Naming this method `similar` seems ambiguous.  It's not clear how similar the notes need to be or what note attributes need to be the same in order to return `true`.  What about using something like `equals` or `equalsTimeAndLength` to be more specific?